packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -189,14 +189,19 @@ export function launchPatientWorkspace(name: string, additionalProps?: object) {
       name,
       additionalProps,
     });
-  } else if (isWorkspaceAlreadyOpen) {
-    openWorkspaces[workspaceIndexInOpenWorkspaces].additionalProps = newWorkspace.additionalProps;
-    const restOfWorkspaces = openWorkspaces.filter((w) => w.name != name);
-    updateStoreWithNewWorkspace(openWorkspaces[workspaceIndexInOpenWorkspaces], restOfWorkspaces);
-  } else if (!!openedWorkspaceWithSameType) {
-    const restOfWorkspaces = store.getState().openWorkspaces.filter((w) => w.type != newWorkspace.type);
-    updateStoreWithNewWorkspace(openedWorkspaceWithSameType, restOfWorkspaces);
-    promptBeforeLaunchingWorkspace(openedWorkspaceWithSameType, {
+  } else if (isWorkspaceAlreadyOpen || !!openedWorkspaceWithSameType) {
+    const workspaceToApply: OpenWorkspace = !!openedWorkspaceWithSameType
+      ? openedWorkspaceWithSameType
+      : {
+        ...openWorkspaces[workspaceIndexInOpenWorkspaces],
+        additionalProps: newWorkspace.additionalProps
+      };
+    const restOfWorkspaces = !!openedWorkspaceWithSameType
+      ? openWorkspaces.filter((w) => w.type != newWorkspace.type)
+      : openWorkspaces.filter((w) => w.name != name);
+
+    updateStoreWithNewWorkspace(workspaceToApply, restOfWorkspaces);
+    promptBeforeLaunchingWorkspace(workspaceToApply, {
```

Comment:
So, a few things:

1. It's preferable (I think) to leave the existing logic alone and add a new else clause that handles the case where a workspace of the same type is already open.
2. The prompting needs to happen _before_ any changes to the store and if the prompt is cancelled, no state updates should be made.
3. Please verify that all existing tests pass and, if necessary, update the failing tests to reflect the new flow.

Reply:
Hi Ian, there is already a flow handled if a workspace of the same type is already opened. There is a prompt whenever user tries to open another workspace of same type.

Reply:
How should this work move forward, @vasharma05? Could you give @elimm direction on how to leverage the existing functionality to achieve the goals @ibacher outlined above?

Reply:
@vasharma05 This _is_ the code that implements that flow. What @elimm is trying to add here is the ability to open _the same workspace_ with new settings, basically to be able to edit an existing encounter when a form is open in the workspace, but not enough attention was paid to the ensure that the tests that rely on the existing behavior pass.

Reply:
updated the tests, made the state change dependent on the result of the prompt